docs/reference/api_references.rst

```diff
@@ -299,12 +299,17 @@ cms.app_base
 
     .. method:: get_menus(page, language, **kwargs)
 
-            Returns the menus for the apphook instance, eventually selected according
+        .. versionadded:: 3.3
+            ``CMSApp.get_menus`` accepts page, language and generic keyword arguments:
+            you can customize this function to return different list of menu classes
+            according to the given arguments.
+
+            Returns the menus for the apphook instance, selected according
             to the given arguments.
 
-            By default it returns the menus assigned to :py:attr:`CMSApp._menus`
+            By default it returns the menus assigned to :py:attr:`CMSApp._menus`.
 
-            If no page and language si provided, this method **must** return all the
+            If no page and language are provided, this method **must** return all the
             menus used by this apphook. Example::
```

Comment:
@yakky @mkoistinen Actually, if I replace:

    urls = ["polls.urls"]

with:

    def get_urls(self, page=None, language=None, **kwargs):
        return ["polls.urls"]

in an apphook, it simply does not work:

``'NoneType' object is not iterable`` because ``get_app_urls(app.urls)`` returns ``[]``.

Any ideas?